Simplenote/TagAttachmentCell.swift

```diff
@@ -0,0 +1,135 @@
+import Foundation
+
+
+// MARK: - TagAttachmentCell
+//
+class TagAttachmentCell: NSTextAttachmentCell {
+
+    /// Text Color
+    ///
+    var textColor: NSColor = .simplenoteTextColor
+
+    /// Font
+    ///
+    var textFont: NSFont = .simplenoteSecondaryTextFont
+
+    /// Returns the receiver's StringValue as a Foundation String
+    ///
+    private var nsStringValue: NSString {
+        stringValue as NSString
+    }
+
+    override func cellSize() -> NSSize {
+        let textSize    = nsStringValue.size(withAttributes: attributes)
+        let width       = textSize.width.rounded(.up) + Metrics.textInsets.left + Metrics.textInsets.right + Metrics.bgInsets.left + Metrics.bgInsets.right
+        let height      = textSize.height.rounded(.up) + Metrics.textInsets.top + Metrics.textInsets.bottom + Metrics.bgInsets.top
+
+        return NSSize(width: width, height: height)
+    }
+
+    override func cellBaselineOffset() -> NSPoint {
+        guard let font = font else {
+            return .zero
+        }
+
+        let offsetY = font.descender.rounded(.down) - Metrics.textInsets.bottom
+        return NSPoint(x: .zero, y: offsetY)
+    }
+
+    override func draw(withFrame cellFrame: NSRect, in controlView: NSView?) {
+        drawBackground(in: cellFrame)
+        drawText(in: cellFrame)
+    }
+
+    override func draw(withFrame cellFrame: NSRect, in controlView: NSView?, characterIndex charIndex: Int, layoutManager: NSLayoutManager) {
```

Comment:
**Now that you mention this!!** I ended up leaving the first implementation that I managed to get working.

Both API(s) are since 10.10 ([Here](https://developer.apple.com/documentation/appkit/nstextattachmentcellprotocol/1508412-draw) and [here](https://developer.apple.com/documentation/appkit/nstextattachmentcellprotocol/1508385-draw).

As far as I understand, the SDK simply calls the "simplest override available" (extract from NSTextAttachment.h):

```
The class NSTextAttachmentCell implements them to simply call the simpler methods
```

In the aims of... keeping things as simple as possible, I'm switching to the API that doesn't receive a LayoutManager reference (since we don't need it). And adding notes to detail exactly when `draw` method runs (Edition or Display modes).

**Thank you Eric!!**